front/src/applications/operationalStudies/hooks/useSimulationResults.ts

```diff
@@ -26,154 +27,120 @@ import type { SimulationResultsData } from '../types';
+    selectedTimetableItemSimulation: Extract<SimulationResponse, { status: 'success' }>;
+    path: PathfindingResultSuccess;
+  }>();
+
+  const [getTrainScheduleById] = osrdEditoastApi.endpoints.getTrainScheduleById.useLazyQuery();
+  const [getPacedTrainById] = osrdEditoastApi.endpoints.getPacedTrainById.useLazyQuery();
+  const [getTrainScheduleByIdPath] =
+    osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useLazyQuery();
+  const [getPacedTrainByIdPath] = osrdEditoastApi.endpoints.getPacedTrainByIdPath.useLazyQuery();
+  const [getTrainScheduleSimulation] =
+    osrdEditoastApi.endpoints.getTrainScheduleByIdSimulation.useLazyQuery();
+  const [getPacedTrainSimulation] =
+    osrdEditoastApi.endpoints.getPacedTrainByIdSimulation.useLazyQuery();
+
+  const fetchSimulationResults = useCallback(async (trainId: TrainId) => {
+    if (!trainId) {
+      setSimulationResults(undefined);
+      return;
     }
-    const pacedTrainId = extractPacedTrainIdFromOccurrenceId(selectedTrainId);
-    return extractEditoastIdFromPacedTrainId(pacedTrainId);
-  }, [selectedTrainId]);
 
-  const { data: selectedTrainSchedule } = osrdEditoastApi.endpoints.getTrainScheduleById.useQuery(
-    {
-      id: editoastSelectedTrainId!,
-    },
-    { skip: !editoastSelectedTrainId || (selectedTrainId && !isTrainScheduleId(selectedTrainId)) }
-  );
-
-  const { data: selectedPacedTrain } = osrdEditoastApi.endpoints.getPacedTrainById.useQuery(
-    {
-      id: editoastSelectedTrainId!,
-    },
-    { skip: !editoastSelectedTrainId || (selectedTrainId && !isOccurrenceId(selectedTrainId)) }
-  );
+    let path: PathfindingResult;
+    let selectedTimetableItem: TimetableItem;
+    let selectedTimetableItemSimulation: SimulationResponse;
 
-  const { data: rawTrainSchedulePath } =
-    osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
-      {
-        id: editoastSelectedTrainId!,
-        infraId: infraId!,
-      },
-      {
-        skip:
-          !editoastSelectedTrainId ||
-          !infraId ||
-          (selectedTrainId && !isTrainScheduleId(selectedTrainId)),
-      }
-    );
-
-  const { data: rawPacedTrainPath } = osrdEditoastApi.endpoints.getPacedTrainByIdPath.useQuery(
-    {
-      id: editoastSelectedTrainId!,
-      infraId: infraId!,
-    },
-    {
-      skip:
-        !editoastSelectedTrainId ||
-        !infraId ||
-        (selectedTrainId && !isOccurrenceId(selectedTrainId)),
+    if (isTrainScheduleId(trainId)) {
+      const rawTrainScheduleId = extractEditoastIdFromTrainScheduleId(trainId);
+      const trainSchedule = await getTrainScheduleById({ id: rawTrainScheduleId }).unwrap();
+      const trainScheduleSimulation = await getTrainScheduleSimulation({
+        id: rawTrainScheduleId,
+        infraId,
+        electricalProfileSetId,
+      }).unwrap();
+
+      selectedTimetableItem = {
+        ...trainSchedule,
+        id: trainId,
+      };
+      selectedTimetableItemSimulation = trainScheduleSimulation;
```

Comment:
nit: we could do that directly ? (same for the paced train case)
```suggestion
      selectedTimetableItemSimulation= await getTrainScheduleSimulation({
        id: rawTrainScheduleId,
        infraId,
        electricalProfileSetId,
      }).unwrap();

      selectedTimetableItem = {
        ...trainSchedule,
        id: trainId,
      };
```

Reply:
Done